W-18929191-Private-Link-Connection-LDS

luanamulesoft opened the pull request.

luanamulesoft committed:
W-18929191-Private-Link-Connection-LDS

luanamulesoft committed:
moved file

luanamulesoft committed:
fixed format

luanamulesoft committed:
typo

rithishapadmanabh commented on cloudhub-2/modules/ROOT/pages/ps-outbound-private-link.adoc:
```diff
@@ -0,0 +1,113 @@
++
+. Set up VPC endpoint services (if self-hosted endpoint services)
+.. https://docs.aws.amazon.com/vpc/latest/privatelink/create-endpoint-service.html[Create a service powered by AWS PrivateLink]
+.. If the service is hosted the same region as the private space, set up the services in the same availability zone as the private space (obtained in he first step here)
```
obtained in he first -- typo

rithishapadmanabh commented on cloudhub-2/modules/ROOT/pages/ps-outbound-private-link.adoc:
```diff
@@ -0,0 +1,113 @@
+.. In the VPC *Endpoint services*, click *Allow principals*
+... Add `arn:aws:iam::{accountId}:root` as principal
++
+The `{accountId}` is obtained in the first step step here
```
first step step here - typo
Is 'here' supposed to be hyperlinked

rithishapadmanabh commented on cloudhub-2/modules/ROOT/pages/ps-outbound-private-link.adoc:
```diff
@@ -0,0 +1,113 @@
+== Limitations
+
+* This configuration supports interface endpoints only
+* This configuration doesn't support CloudHub VPCs or CloudHub 2.0 migrated private spaces
```
I would split into 2 points.
* Does not support CloudHub VPCs
* Does not support CloudHub 2.0 Private Spaces that were migrated or upgraded from CloudHub (hyperlink our VPC upgrade docs here). These private spaces will be able to leverage the feature only after the migration is completed and the underlying CloudHub VPC is deleted.

rithishapadmanabh commented on cloudhub-2/modules/ROOT/pages/ps-outbound-private-link.adoc:
```diff
@@ -0,0 +1,113 @@
+
+* This configuration supports interface endpoints only
+* This configuration doesn't support CloudHub VPCs or CloudHub 2.0 migrated private spaces
+* You can't perform cross-region validation properly done at the API level. If the endpoint service is invalid, the VPCE provisioning fails.
```
I don't follow this point/limitation - "you can't perform cross-region validation properly done at the API"
Need to clarify

xuanshi replied:
this is not customer facing. It's the limitation of our product.

rithishapadmanabh commented on cloudhub-2/modules/ROOT/pages/ps-outbound-private-link.adoc:
```diff
@@ -0,0 +1,113 @@
+* This configuration supports interface endpoints only
+* This configuration doesn't support CloudHub VPCs or CloudHub 2.0 migrated private spaces
+* You can't perform cross-region validation properly done at the API level. If the endpoint service is invalid, the VPCE provisioning fails.
+
```
We need a section on entitlements - each VPC endpoint will consume one of your (customer's) network connection (in the new P&P) and VPN (in the legacy vcore p&p) entitlements.

rithishapadmanabh commented on cloudhub-2/modules/ROOT/pages/ps-outbound-private-link.adoc:
```diff
@@ -0,0 +1,113 @@
+----
+
+
+== Limitations
```
Also need to add a limitation on max rate limit per org per hour similar to private connect:

"Outbound PrivateLink connections have a contractual limit of 56.48 GB of data transferred per root org per hour."
Similar to : https://help.salesforce.com/s/articleView?id=xcloud.private_connect_considerations.htm&type=5

rithishapadmanabh replied:
@xuanshi Also need another limitation around number of connections/endpoints per private space limit. I believe our external docs puts it as VPNs per private space = 10 and TGW per private space = 5.

We also have 
Connections per private space
5. 10 if the connections are not redundant.

- not sure if this is network connections or something else (perhaps not network connections (which can be both vpn/tgw and in this case privatelink), since 5 is < 10 which is the limit for VPN)

rithishapadmanabh commented on cloudhub-2/modules/ROOT/pages/ps-outbound-private-link.adoc:
```diff
@@ -0,0 +1,113 @@
+* This configuration doesn't support CloudHub VPCs or CloudHub 2.0 migrated private spaces
+* You can't perform cross-region validation properly done at the API level. If the endpoint service is invalid, the VPCE provisioning fails.
+
+
```
@xuanshi -
1. Can we document prerequisites on number of IPs that need to be free in the private space for each VPC endpoint.
2. Can we document how can customers configure high availability - would that consume one vpc endpoint (network connection) per AZ (total 3) or would it be a single one

rithishapadmanabh reviewed (dismissed):
Need added content on limitations, entitlements, prerequisites + some typos/corrections on clarity

xuanshi commented on cloudhub-2/modules/ROOT/pages/ps-outbound-private-link.adoc:
```diff
@@ -0,0 +1,113 @@
+= Setting Up an Outbound Private Link Connection
+
+This document describes the steps to configure and outbbound private link connection from CloudHub 2.0 private spaces.
```
an outbound?

luanamulesoft committed:
applied reviews

luanamulesoft committed:
Update ps-outbound-private-link.adoc

luanamulesoft committed:
Update ps-outbound-private-link.adoc

luanamulesoft committed:
Update ps-outbound-private-link.adoc

luanamulesoft committed:
Update ps-outbound-private-link.adoc

luanamulesoft committed:
Update ps-outbound-private-link.adoc

luanamulesoft committed:
Update ps-outbound-private-link.adoc

luanamulesoft committed:
Update ps-outbound-private-link.adoc

luanamulesoft committed:
Delete .github/copilot-instructions.md

luanamulesoft committed:
Update ps-outbound-private-link.adoc

luanamulesoft committed:
Update ps-outbound-private-link.adoc

luanamulesoft committed:
Merge branch 'latest' into W-18929191-Private-Link-Connection-LDS

luanamulesoft committed:
Update ps-outbound-private-link.adoc

luanamulesoft committed:
Update ps-outbound-private-link.adoc

luanamulesoft committed:
Merge branch 'latest' into W-18929191-Private-Link-Connection-LDS

luanamulesoft committed:
Integrated content from blog

luanamulesoft committed:
Update ps-outbound-private-link.adoc

luanamulesoft committed:
capitalization

luanamulesoft committed:
Update ps-outbound-private-link.adoc

luanamulesoft committed:
added API parameters

luanamulesoft committed:
Update ps-outbound-private-link.adoc

luanamulesoft committed:
Update ps-outbound-private-link.adoc

luanamulesoft committed:
Update ps-outbound-private-link.adoc

luanamulesoft committed:
Update ps-outbound-private-link.adoc

luanamulesoft committed:
Update ps-outbound-private-link.adoc

luanamulesoft committed:
Update ps-outbound-private-link.adoc

luanamulesoft committed:
Update ps-outbound-private-link.adoc

luanamulesoft committed:
Update ps-outbound-private-link.adoc

luanamulesoft committed:
Update ps-outbound-private-link.adoc

luanamulesoft committed:
Update ps-outbound-private-link.adoc

luanamulesoft committed:
Update ps-outbound-private-link.adoc

luanamulesoft committed:
Merge branch 'latest' into W-18929191-Private-Link-Connection-LDS

luanamulesoft committed:
Merge branch 'latest' into W-18929191-Private-Link-Connection-LDS

luanamulesoft committed:
Add countOfAzs parameter documentation for VPC endpoint API

- Document countOfAzs as alternative to azIds for POST/PATCH endpoints
- Specify valid values as 1 or 2
- Add example showing countOfAzs usage in S3 endpoint section
- Clarify that CloudHub 2.0 auto-selects AZs when countOfAzs is used

luanamulesoft committed:
Merge branch 'W-18929191-Private-Link-Connection-LDS' into W-21037657-private-link-update-LDS

luanamulesoft committed:
Merge pull request #371 from mulesoft/W-21037657-private-link-update-LDS

W-21037657-private-link-update-LDS

hannanelson committed:
Merge branch 'latest' into W-18929191-Private-Link-Connection-LDS
